include/logger/loggerConfig.h

```diff
@@ -272,59 +276,104 @@ typedef struct _PWMConfig {
 #define DEFAULT_PWM_CONFIG {DEFAULT_PWM_CHANNEL_CONFIG, MODE_PWM_FREQUENCY, MODE_LOGGING_PWM_DUTY, DEFAULT_PWM_DUTY_CYCLE, DEFAULT_PWM_PERIOD}
 
 
-#define OBD2_CHANNELS 20
+#define DEFAULT_CAN_BAUD_RATE 500000
+
+typedef struct _CANConfig {
+    unsigned char enabled;
+    int baud[CONFIG_CAN_CHANNELS];
+#if CAN_SW_TERMINATION == true
+    bool termination[CONFIG_CAN_CHANNELS];
+#endif
+} CANConfig;
+
+/* define max offsets and length for CAN mappings */
+#define MAX_CAN_MAPPING_OFFSET_BYTES 7
+#define MAX_CAN_MAPPING_LENGTH_BYTES 4
+
+enum CANMappingEndian {
+    CANMappingEndian_Big = 0,
+    CANMappingEndian_Little
+};
+
+
+typedef struct _CANMapping {
+    /* The standard channel configuration */
+    ChannelConfig channel_cfg;
+
+    /* CAN ID to match this mapping */
+    uint32_t can_id;
+
+    /* mask to apply to ID */
+    uint32_t can_mask;
+
+    /* multiplier for conversion formula */
+    float multiplier;
+
+    /* divider for conversion formula */
+    float divider;
+
+    /* adder for conversion formula */
+    float adder;
+
+    /* flag to indicate bit mode or byte mode for offset and length */
+    uint8_t bit_mode;
+
+    /* indicates the encoding of the raw data type: unsigned, signed, IEEE 754 floating point */
+    uint8_t type;
+
+    /* the can bus we expect the data to show up on */
+    uint8_t can_channel;
+
+    /* indicates endian-ness for multi-byte values */
+    uint8_t big_endian;
+
+    /* byte or bit offset of the data to extract within the CAN message */
+    uint8_t offset;
+
+    /* byte or bit length of the data to extract within the CAN message */
+    uint8_t length;
+
+    /* the conversion filter to apply to quickly convert units without changing mapping */
+    uint8_t conversion_filter_id;
+} CANMapping;
+
+typedef struct _CANChannel {
+    /* The standard channel configuration */
+    ChannelConfig channel_cfg;
+    CANMapping mapping;
+} CANChannel;
+
+typedef struct _CANChannelConfig {
+    /* all available CAN mappings */
+    CANChannel can_channels[CONFIG_CAN_MAPPINGS];
+    /* globally enables/disables CAN mappings */
+    uint8_t enabled;
+    /* number of mappings set within configuration */
+    uint16_t enabled_mappings;
+} CANChannelConfig;
+
 
 typedef struct _PidConfig {
     ChannelConfig cfg;
-    unsigned short pid;
+    CANMapping mapping;
+
+    /* flag for passive mode where we only listen for OBDII PID responses */
+    uint8_t passive;
+
+    /* The OBDII PID to query */
+    uint8_t pid;
+
+    /* The OBDII mode specified in the query */
+    uint16_t mode;
```

Comment:
I was unsure, but I'm defaulting to not needing an enum currently - there are some standard modes - like 01 - 09 and then mode 0x22, but it seems anything could be specified. So I'm leaving it as  a regular number.